src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1447,26 +1461,115 @@ protected override void AfterLeftChildHasBeenVisited(BoundBinaryOperator binary)
                             {
-                                if (slot >= this.State.Capacity) Normalize(ref this.State);
-
-                                Split();
+                                // If we got a slot we must have processed the previous conditional receiver.
+                                Debug.Assert(_lastConditionalAccessSlot == -1);
 
-                                if (op == BinaryOperatorKind.Equal)
+                                if (slot >= this.State.Capacity) Normalize(ref this.State);
+                                if (slotListBuilder == null)
                                 {
-                                    this.StateWhenFalse[slot] = true;
+                                    slotListBuilder = ArrayBuilder<int>.GetInstance();
                                 }
-                                else
+                                slotListBuilder.Add(slot);
+
+                                // When MakeSlot is called on the nested AccessExpression, it will recurse through receivers
+                                // until it gets to the BoundConditionalRecevier associated with this node. In our override
+                                _lastConditionalAccessSlot = slot;
+
+                                operand = conditional.AccessExpression;
+                                continue;
+                            }
+                            else
+                            {
+                                // If we didn't get a slot, it's possible that the current _lastConditionalSlot was never processed,
+                                // so we reset before leaving the function.
+                                _lastConditionalAccessSlot = -1;
+
+                                // If the current element doesn't have a slot, nothing underneath will have a slot either.
+                                return slotListBuilder?.ToImmutableAndFree() ?? ImmutableArray<int>.Empty;
+                            }
+                        default:
+                            // Attempt to create a slot for the current thing. If there were any more conditional accesses,
+                            // they would have been on top, so this is the last thing we need to specially handle.
+
+                            // PROTOTYPE(NullableReferenceTypes): When we handle unconditional access survival (ie after
+                            // c.D has been invoked, c must be nonnull or we've thrown a NullRef), revisit whether
+                            // we need more special handling here
+
+                            slot = MakeSlot(operand);
+                            if (slot > 0)
+                            {
+                                // If we got a slot we must have processed the previous conditional receiver.
+                                Debug.Assert(_lastConditionalAccessSlot == -1);
+
+                                if (slot >= this.State.Capacity) Normalize(ref this.State);
```

Comment:
>Normalize(ref this.State); [](start = 65, length = 26)

Consider calling `Normalize` once, unconditionally, in the caller of `getOperandSlots`.